### ISSUE TITLE
Monitor option is only available for single LR scheduler

### ISSUE THREAD

**Issue by johngull** [PyTorchLightning/pytorch-lightning#6617]
## 🐛 Bug

I'm trying to use LR schedulers for both generator and discriminator in GAN by the code:

```
def configure_optimizers(self,):
....
return [
            {
            'optimizer': optimizer,
            'lr_scheduler': lr_scheduler,
            'monitor': monitor,
            'interval': 'epoch' if reduce else 'step',
            'reduce_on_plateau': reduce
            }
            for optimizer, lr_scheduler in zip(optimizers, lr_schedulers)
        ]
```

But for `ReduceLROnPlateau` I got an exception that monitor value can't be None.

Actually, I found that `TrainerOptimizersMixin.init_optimizers` handle monitor value only for the case of a single dictionary and no other cases.

Obviously, `monitor` value can be passed in the case of multiple dictionaries also and should be handled in this case also.

**Comment by carmocca** [comment]
Re-labeling as an enhancement as this is more about better support for multiple schedulers than something not working.

Would you like to work on a PR for this?

**Comment by SkafteNicki** [comment]
I just think that your return statement should look different:
``` python
def configure_optimizers(self):
    ...
    return optimizers, [{'scheduler': sch, 'monitor': monitor, 'interval': 'epoch' if reduce else 'step'} for sch in scheduler]
```
instead of list of dicts, return two list: one of optimizers and one for schedulers. The scheduler list then consist of dicts for each scheduler.
you can see more in the docs: https://pytorch-lightning.readthedocs.io/en/stable/common/optimizers.html#learning-rate-scheduling

**Comment by johngull** [comment]
@SkafteNicki docs of pytorch-lightning state that both ways are valid.

And your example will also fail with ReduceLROnPlateau inside the list of dicts, what is the main point of the issue.
